### PR TITLE
Prevent elasticsearch from reindexing when  source index matches destination index

### PR DESCRIPTION
**Pull request by flamingbear**
**Summary:** Summary of changes

Addresses [CUMULUS-2397: Prevent Reindex operation to the same index](https://bugs.earthdata.nasa.gov/browse/CUMULUS-2397)

## Changes

* elasticsearch endpoint throws error when destination index is equal to the source index. 

## PR Checklist

- [ ] Update CHANGELOG
- [ X ] Unit tests
- [ X ] Adhoc testing
- [ n/a ] Integration tests

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -34,15 +34,25 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
...
     indicating that the endpoint has been disabled.
 - **CUMULUS-2355**
-  - Added logic to disable `/s3Credentials` endpoint based upon value for environment variable `DISABLE_S3_CREDENTIALS`. If set to "true",  the endpoint will not dispense S3 credentials and instead return a message indicating that the endpoint has been disabled.
+  - Added logic to disable `/s3Credentials` endpoint based upon value for
+    environment variable `DISABLE_S3_CREDENTIALS`. If set to "true", the
+    endpoint will not dispense S3 credentials and instead return a message
+    indicating that the endpoint has been disabled.
+- **CUMULUS-2397**
+  - Updated `/elasticsearch` endpoint's `reindex` function to prevent
+    reindexing when source and destination indices are the same.
```
Comment (flamingbear) [inline CHANGELOG.md:51]: My changes, I wrapped my old changes.

File: packages/api/tests/endpoints/test-elasticsearch.js [hunk 2]
```diff
@@ -222,6 +222,24 @@ test.serial('Reindex - specify a source index that is not aliased', async (t) =>
   await esClient.indices.delete({ index: destIndex });
 });
 
+test.serial('Reindex fails when source index matches destination index.', async (t) => {
```
Comment (Jkovarik) [inline packages/api/tests/endpoints/test-elasticsearch.js:225]: 🐜 - slight preference for expanding on 'fail' when not something like `t.throws`

```suggestion
test.serial('Reindex request returns 400 with the expected message when source index matches destination index.', async (t) => {
```

File: packages/api/endpoints/elasticsearch.js [hunk 3]
```diff
@@ -96,6 +96,10 @@ async function reindex(req, res) {
     }
   }
 
+  if (sourceIndex === destIndex) {
```
Comment (Jkovarik) [inline packages/api/endpoints/elasticsearch.js:99]: Given there's logic above around L87/75 checking the existence of the the source/destination index, should this be up around L60?

Comment (Jkovarik) [reply packages/api/endpoints/elasticsearch.js:99]: Hmm, apologies, the logic that prompted the question was the block after source/dest indexes are set.   That comment shouldn't read L60 -- L85 would be :100: though.

Comment (Jkovarik) [reply packages/api/endpoints/elasticsearch.js]: 0646b1af673124fccdb4ecaffb21d14aee3ddb01  👍🏻

Comment (Jkovarik) [reply packages/api/endpoints/elasticsearch.js]: Thanks for the update, and the added unit looks good.

**Review by Jkovarik — APPROVED** [review]
Great work!  👍🏻

**Pull request merged**